src/bin/sage-env

```diff
@@ -370,20 +370,31 @@ if [ -n "$PYTHONHOME" ]; then
 fi
 
 if [ -n "$SAGE_LOCAL" ]; then
-    # On OS X, test whether "ld-classic" is present in the installed
-    # version of the command-line tools. If so, we add "-ld_classic"
-    # to LD_FLAGS. See #36599.
-    # When the full XCode is installed and in use, for example after
-    # "sudo xcode-select -s /Applications/Xcode.app", then "xcode-select -p"
-    # gives "/Applications/Xcode.app/Contents/Developer", but "ld-classic"
-    # is not in the subdirectory "usr/bin/" but rather in the subdirectory
-    # "Toolchains/XcodeDefault.xctoolchain/usr/bin/". See #37237.
-    # However, if LD is set explicitly, as it is within conda on macOS,
-    # do not not do this.
-    if [ "$UNAME" = "Darwin" ] && [ -z "$LD" ] && [ -x "$(xcode-select -p)/usr/bin/ld-classic" -o -x "$(xcode-select -p)/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld-classic" ] ; then
-        LDFLAGS="-L$SAGE_LOCAL/lib -Wl,-ld_classic,-rpath,$SAGE_LOCAL/lib $LDFLAGS"
-    else
-        LDFLAGS="-L$SAGE_LOCAL/lib -Wl,-rpath,$SAGE_LOCAL/lib $LDFLAGS"
+    # On OS X, test whether "ld-classic" is present in the selected XCode
+    # toolchain. If so, add "-Wl,-ld_classic" to LDFLAGS (see #36599) unless
+    # LD is already set, as it will be with conda on macOS.  When the selected
+    # toolchain is in the Xcode app the output of "xcode-select -p" is
+    # "/Applications/Xcode.app/Contents/Developer", but "ld-classic" is not in
+    # the subdirectory "usr/bin/" but rather in the subdirectory
+    # "Toolchains/XcodeDefault.xctoolchain/usr/bin/".  (See #37237.)
+    LDFLAGS="-L$SAGE_LOCAL/lib $LDFLAGS"
+    if [ "$UNAME" = "Darwin" ] && [ -z "$LD" ]; then
+	# Running xcode-select on a system with no toolchain writes an
+	# error message to stderr, so redirect stderr to /dev/null. 
+	XCODE_PATH=$(/usr/bin/xcode-select -p 2> /dev/null)
+	if [ -n $XCODE_PATH ]; then
+            if [ -x "$XCODE_PATH/usr/bin/ld-classic" -o \
+		 -x "$XCODE_PATH/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld-classic" ]; then
+		LDFLAGS="$LDFLAGS -Wl,-ld_classic"
+            fi
+        else
+            # On a macOS system with no toolchain we don't want this script
+            # to call gcc because that will also print an error message to
+            # stderr.  We can avoid this by setting AS and LD to their
+            # default values.
+            AS=as
+            LD=ld
+	fi
     fi
     if [ "$UNAME" = "Linux" ]; then
         LDFLAGS="-Wl,-rpath-link,$SAGE_LOCAL/lib $LDFLAGS"
```

Comment:
...with rpath-link breaks various programs, see 37982

Reply:
I can't find either an issue or a pull request with number 37982 so I have no idea what the problem is.  Can you provide a link please?